pom.xml

```diff
@@ -749,6 +755,7 @@
                 <id>attach-sources</id>
                 <goals>
                   <goal>jar</goal>
+                  <goal>test-jar</goal> <!-- TODO ideally would be skipped unless we have a test-jar execution of maven-jar-plugin -->
```

Comment:
Can try that. My initial attempt at using a profile failed, but I think what you suggest could be made to work, since there is also a `skip` parameter for `jar:test-jar`.